.github/CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,128 @@
+or harmful.
+
+Community leaders have the right and responsibility to remove, edit, or reject
+comments, commits, code, wiki edits, issues, and other contributions that are
+not aligned to this Code of Conduct, and will communicate reasons for moderation
+decisions when appropriate.
+
+## Scope
+
+This Code of Conduct applies within all community spaces, and also applies when
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+https://discord.gg/ZeuASSx.
+All complaints will be reviewed and investigated promptly and fairly.
+
+All community leaders are obligated to respect the privacy and security of the
+reporter of any incident.
+
+## Enforcement Guidelines
+
+Community leaders will follow these Community Impact Guidelines in determining
+the consequences for any action they deem in violation of this Code of Conduct:
+
+### 1. Correction
+
+**Community Impact**: Use of inappropriate language or other behavior deemed
+unprofessional or unwelcome in the community.
+
+**Consequence**: A private, written warning from community leaders, providing
+clarity around the nature of the violation and an explanation of why the
+behavior was inappropriate. A public apology may be requested.
+
+### 2. Warning
+
+**Community Impact**: A violation through a single incident or series
+of actions.
+
+**Consequence**: A warning with consequences for continued behavior. No
+interaction with the people involved, including unsolicited interaction with
+those enforcing the Code of Conduct, for a specified period of time. This
+includes avoiding interactions in community spaces as well as external channels
+like social media. Violating these terms may lead to a temporary or
+permanent ban.
+
+### 3. Temporary Ban
+
+**Community Impact**: A serious violation of community standards, including
+sustained inappropriate behavior.
+
+**Consequence**: A temporary ban from any sort of interaction or public
+communication with the community for a specified period of time. No public or
+private interaction with the people involved, including unsolicited interaction
+with those enforcing the Code of Conduct, is allowed during this period.
+Violating these terms may lead to a permanent ban.
+
+### 4. Permanent Ban
+
+**Community Impact**: Demonstrating a pattern of violation of community
+standards, including sustained inappropriate behavior,  harassment of an
+individual, or aggression toward or disparagement of classes of individuals.
+
+**Consequence**: A permanent ban from any sort of public interaction within
+the community.
```

Comment:
This doesn't really feel clear to me.

Temporary ban says:
- if you ignore the ban and try to interact with members of the community, especially in a negative way, you might receive a permanent ban.

If they've managed to break free from their temporary ban, though, how are you going to enforce a permanent ban? Obviously, this probably involves the offender getting banned from the Discord, blocked from interacting with the GitHub page or the dev team, etc. But if they've already circumvented the temporary ban (which appears to be exactly the same as the permanent ban, just temporary), what's going to stop them from doing the same here? Is there going to be some action taken with outside support from GitHub or Discord?

I've got even more questions, actually:
- What even constitutes "public interaction?"
- Who among the moderators has the authority to turn a temporary ban into a permanent one? If moderator X is being "harassed" by temp-banned user Y when really Y just has poor communication skills/didn't properly read the code of conduct and is trying to get their ban appealed, should moderator X be able to execute this somewhat drastic step? Is a team of two or three moderators needed?
- The criteria for a permanent ban seem to be broader and no more serious than those for a temporary ban. The only difference is the "serious violation" vs "pattern of violation." What makes the permanent ban more severe beyond it being applied more sparingly and at the moderators' discretion?

Reply:
Public interaction would in this case mean posting or commenting on things in the repo.

I would say that whichever moderator applies the temporary ban is not allowed to be the same moderator who applies the permanent on. That's probably as much insurance as we can reasonably build into that though.

As for what makes it more severe?
It really just comes down to whether a person is actually capable of taking on board feedback and trying to do something about their behaviour. The temporary ban serves as a warning, the permanent ban is the decision for when the warning has no effect.

Reply:
Those explanations should probably be added to the code of conduct in some way.